sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_group_decorator.py

```diff
@@ -152,17 +152,38 @@ def _create_init_fn(cls, fields):  # pylint: disable=unused-argument
         def _get_data_type_from_annotation(anno: Input):
             if isinstance(anno, GroupInput):
                 return anno._group_class
-
-            return IOConstants.PRIMITIVE_STR_2_TYPE[anno.type]
+            # keep original annotation for Outputs
+            if isinstance(anno, Output):
+                return anno
+            try:
+                # convert to primitive type annotation if possible
+                return IOConstants.PRIMITIVE_STR_2_TYPE[anno.type]
+            except KeyError:
+                # otherwise, keep original annotation
+                return anno
```

Comment:
Is the branch for data type annotation? what's the behavior if user write a custom annotation, will it fail somewhere behind later?

Reply:
It will fail with user error when used in @pipeline, see `_pipeline_component_builder.py/_validate_group_annotation`, let me add a test for it.

Reply:
Oh, I remembered that.